src/arrayMethodReduce.js

```diff
@@ -4,8 +4,17 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(accamulatorFunc, initialValue) {
+    const hasInitialValue = arguments.length > 1;
+    let accamulator = hasInitialValue ? initialValue : this[0];
+
+    for (let i = hasInitialValue ? 0 : 1; i < this.length; i++) {
+      if (i in this) {
```

Comment:
Why do you need "i in this" check? Everything seems to work fine without it.

Reply:
Thank you! Made all the changes!